src/resolvers/FileSystemResolver.ts

```diff
@@ -0,0 +1,54 @@
+import { stat } from 'mz/fs';
+import { resolve } from 'path';
+import Resolver from './Resolver';
+
+const DEFAULT_PLUGIN_EXTENSIONS = new Set(['.js']);
+
+async function isFile(path: string): Promise<boolean> {
+  try {
+    return (await stat(path)).isFile();
+  } catch {
+    return false;
+  }
+}
+
+/**
+ * Resolves file system paths to plugins.
+ */
+export default class FileSystemResolver implements Resolver {
+  constructor(
+    private readonly optionalExtensions: Set<string> = DEFAULT_PLUGIN_EXTENSIONS
+  ) {}
+
+  private *enumerateCandidateSources(source: string): IterableIterator<string> {
+    yield resolve(source);
```

Comment:
I think I misunderstood the flow in this function.  `optionalExtensions` is used here so that the user can provide a path `path/to/filename.js` or `path/to/filename`, where currently if extension is omitted, we will try to resolve using this extension array. Is that correct?

Reply:
Yes, that's right.